packages/forms/src/UIForm/Widget/condition.js

```diff
@@ -1,29 +1,172 @@
-				return value.length;
+	if (typeof value === 'number') {
+		return JSON.stringify(value);
+	}
+	if (typeof value === 'string') {
+		return value;
+	}
+	return onError(value);
+}
+
+/**
+ * "foo=bar;dummy=1" will return [{foo:"bar",dummy:"1"}]
+ */
+function parseParameters(parameterString) {
+	return parameterString
+		.split(';')
+		.map(it => it.trim())
+		.filter(it => it.length > 0)
+		.map(it => {
+			const sep = it.indexOf('=');
+			if (sep > 0) {
+				return {
+					[it.substring(0, sep).trim()]: it.substring(sep + 1, it.length).trim(),
+				};
 			}
-			return 0;
-		default:
-			return false;
+			return {
+				value: it.trim(),
+			};
+		})
+		.reduce(
+			(a, v) => ({
+				...a,
+				...v,
+			}),
+			{},
+		);
+}
+
+/**
+ * parses a strategy based on this pattern "name(parameters)" and returns
+ * an object containing { name, params }.
+ */
+function parseStrategy(strategy) {
+	if (!strategy) {
+		return strategy;
 	}
+
+	const strategyPattern = /^([a-zA-Z]{1}\w*)(?:\((.+)?\))?$/;
+	const matches = strategy.match(strategyPattern);
+	const isMatching = matches !== undefined;
+
+	const name = isMatching ? matches[1] : strategy;
+	const parameters = isMatching ? matches[2] : undefined;
+
+	const hasParameters = parameters !== undefined;
+
+	return {
+		name: name.toLowerCase(),
+		params: hasParameters ? parseParameters(parameters) : {},
+	};
 }
 
+function areEqualsAsNumbers(expected, actualStringOrNumber) {
+	return expected === stringOrNumberToNumber(actualStringOrNumber);
+}
+
+function containsString(expected, actualStringOrNumber, valueProcessor = v => v) {
+	return (
+		actualStringOrNumber &&
+		valueProcessor(stringOrNumberToString(actualStringOrNumber)).includes(expected)
+	);
+}
+
+// this will not fail in case actualStringOrNumber is not a string or number
+// but will return false sinc eit is used as a fallback comparison using coercing
+function areEqualsAsString(expected, actualStringOrNumber) {
+	return actualStringOrNumber && stringOrNumberToString(actualStringOrNumber, v => v) === expected;
+}
+
+/**
+ * For an array or a string if compares the expected value to its length.
+ */
+function toLengthEvaluator(value) {
+	if (value && value.length) {
+		const length = value.length;
+		return expected => areEqualsAsNumbers(length, expected);
+	}
+	return expected => areEqualsAsNumbers(0, expected);
+}
+
+/**
+ * If an array it checks if the expected element is present in the array.
+ * If a string it checks the expected token is present in the string.
+ *
+ * It accepts the <code>lowercase</code> option to do a test using the
+ * lowercased value of the actual value.
+ */
+function toContainsEvaluator(actual, options) {
+	if (!actual) {
+		return () => false;
+	}
+	if (Array.isArray(actual)) {
+		if (options.lowercase === 'true') {
+			return expected => actual.map(it => it.toLowerCase()).indexOf(expected) >= 0;
+		}
+		return expected => actual.indexOf(expected) >= 0;
+	}
+	if (options.lowercase === 'true') {
```

Comment:
Please don't use `'true'` as a boolean expression. It should be normalized beforehand.